mqtt/mqtt-broker/src/broker.rs

```diff
@@ -306,6 +306,8 @@ where
                 debug!("dropping connection due to connect packet error");
                 let message = Message::Client(client_id, ClientEvent::DropConnection);
                 try_send!(connreq.handle_mut(), message);
+
+                info!("client connection refused");
```

Comment:
Is this useful? Do you want to include clientId and reason why it was refused?

Reply:
client_id is in the context of trace event. also before each refuse_connection call, there is a warning message

Reply:
I am not sure i follow.. 
> client_id is in the context of trace event.
Isn't this change to address the case when only info logging is enabled?
>also before each refuse_connection call, there is a warning message
and that has the info? Why do we still need this then?

Reply:
https://github.com/Azure/iotedge/pull/4525/files#diff-c37c4acce39447d43b2fbcf9c1698aceee26bc6e065afb0f165b85a89f2be923L334-L341

1. before each `refuse_connection!` call we have a warn message explaining what went wrong. so I don't think we need to provide refuse connection message again with info level.
2. since we are using tracing a not logging we can use trace event context and put additional data into it like client_id of the client which packet we process. so we don't need to pass client_id to every log message because it is already there

Reply:
Can you please make this `warn!`

Reply:
Or rather why not update the `debug!` log above? (line 306). Also, it would be good to include reason code.